spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/DefaultGcpProjectIdProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.autoconfig;
```

Comment:
Hm. How about `String projectId`-based ctor instead of hard `GcpProperties` dependency?
The other ctor can be empty, so we fallback to the `ServiceOptions.getDefaultProjectId()`.

With all that logic we won't need to have this in the `autoconfig` package and all the non-Boot users would be happy again